salt/fileserver/__init__.py

```diff
@@ -129,6 +129,12 @@ def check_file_list_cache(opts, form, list_cache, w_lock):
                 else:
                     # if filelist does not exists yet, mark it as expired
                     age = opts.get('fileserver_list_cache_time', 20) + 1
+                if age < (opts.get('fileserver_list_cache_time', 20) - 60*60*2):
+                    # Cache is from the future! Warn and mark cache invalid.
+                    # NOTE: We subtract an additional 2h from fileserver_list_cache_time
+                    #       to avoid change between daylight saving time.
+                    log.warning('The file list_cache was created in the future!')
+                    age = opts.get('fileserver_list_cache_time', 20) + 1
```

Comment:
Why not to just change the following `if` condition to this?
```python
if age > 0 and age < opts.get('fileserver_list_cache_time', 20):
```

Reply:
Because then we lose the warning. It would be good to log those case.

Reply:
Ok, but why not to just check that the age is positive? Because the condition "cache in the future" means `age < 0`, isn't it?

Reply:
```if age > (-60*60*2) and age < opts.get('fileserver_list_cache_time', 20):``` might work and keep the 2 hour padding. However we would still need a separate if above to log the warning.

But honestly I am not even sure the fix proposed here is correct at all, it works for my edge case. But I am not super familiar with this area of the code.

Reply:
I completely agree about the warning. My discussion points here are the following.
1. The warning have to be shown each time we see the negative age.
2. We have to treat negative age as an `invalid cache` condition and recreate the cache in that case.

@terminalmage what do you think?

Reply:
BTW, here we're operating the unix timestamp that is seconds from the epoch in UTC so we shouldn't care about timezones and daily saving.